mgmt/FileManager.cc

```diff
@@ -151,19 +153,21 @@ FileManager::getRollbackObj(const char *fileName, Rollback **rbPtr)
 //
 //
 void
-FileManager::fileChanged(const char *fileName, bool incVersion)
+FileManager::fileChanged(const char *fileName, const char *configName, bool incVersion)
 {
   callbackListable *cb;
-  char *filenameCopy;
+  char *filenameCopy, *confignameCopy;
```

Comment:
Why not
```
std::string filenameCopy;
for (...) {
  filenameCopy.assign(fileName);
  (*cb->func)(filenameCopy.c_str(), ...);
  // No free needed.
}
```

Reply:
Even better
```
for (...) {
  (*cb->func)(std::string{filename}.c_str(), ...);
}
```

Reply:
The problem is that c_str() returns a const char * and the call back function expects char *.  Casting away the const seems less than ideal.